FPIS/Views/AddDesignation.cs

```diff
@@ -0,0 +1,184 @@
+        }
+        private void ValidateInputs()
+        {
+            ValidateDesignationName();
+
+        }
+        private void LoadDeparments()
+        {
+            try
+            {
+                AppDbContext dbContext = new();
+                var departmentName = from Department in dbContext.Departments
+                                     select Department;
+                materialComboBoxDepartmentName.DataSource = departmentName.ToList();
+                materialComboBoxDepartmentName.DisplayMember = "DepartmentName";
+                dbContext.Dispose();
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine($"Error Loading Departments: {ex}");
+                MaterialMessageBox.Show(ex.ToString());
+            }
+        }
+
+        private void AddDesignation_Load(object sender, EventArgs e)
+        {
+           LoadDeparments();
+        }   
+        private void CreateDesignation()
+        {
+            btnAddDesignation.Enabled = false;
+            AppDbContext dbContext = new();
+            var department = dbContext.Departments.Where(dpt => dpt.DepartmentName == materialComboBoxDepartmentName.Text).Single();
+            
+            Designation designation = new()
+            {
+                
+                DesignationName = materialTextBoxDesignationName.Text,
+                DepartmentId = department.Id,
+
+            };
+            try
+            {
+                dbContext.Designations.Add(designation);
+                dbContext.SaveChanges();
+
+                MessageBox.Show(
+                    $"\"{designation.DesignationName}\" Designation created successfully under {materialComboBoxDepartmentName.Text} Department.",
+                    "Success",
+                    MessageBoxButtons.OK,
+                    MessageBoxIcon.Information
+                );
+
+                Console.WriteLine($"Create designation {designation}");
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine($"Error creating Department: {ex}");
+                MessageBox.Show(ex.ToString());
+            }
+            finally
+            {
+                btnAddDesignation.Enabled = true;
+                materialTextBoxDesignationName.Text = "";
+                dbContext.Dispose();
+            }
+        }
+
+        private void ConfirmNewDesignation()
+        {
+            DialogResult dialogResult = MaterialMessageBox.Show(
+                $"Do you want to create \"{designationName}\" as a new Designation?",
+                "Confirm",
+                MessageBoxButtons.YesNo,
+                MessageBoxIcon.Question
+                );
+
+            if (dialogResult == DialogResult.Yes)
+            {
+                CreateDesignation();
+            }
+            else if (dialogResult == DialogResult.No)
+            {
+                materialTextBoxDesignationName.Text = "";
+            }
```

Comment:
Same issue here as with the Department review

Reply:
Resolved as well